utils.js

```diff
@@ -1,23 +1,37 @@
+function getTrailerID() { // Requires DOM level access
+    let p = document.location.pathname;
+    if(!document.location.origin === "https://www.youtube.com") return false
+    if(!p.startsWith("/user/")  && !p.startsWith("/channel/"))  return false
+    if(document.getElementsByClassName("ytp-title-link")[1] === undefined) return false
+    let id = document.getElementsByClassName("ytp-title-link")[1].href.split("?v=")[1];
+    return id.length == 11 ? id : false;
+}
+
 function getYouTubeVideoID(url) {
+    if(url === undefined) {
+        if(id = getTrailerID()) return id;
+        url = document.URL;
+    }
     //Attempt to parse url
-    let urlObject = null;
-    try { 
-        urlObject = new URL(url);
-    } catch (e) {      
-        console.error("[SB] Unable to parse URL: " + url);
-        return false
+    if(typeof url === 'string') {
+        try {
+            url = new URL(url);
+        } catch (e) {      
+            console.error("[SB] Unable to parse URL: " + url);
+            return false
+        }
     }
   
     //Check if valid hostname
-    if(!["www.youtube.com","www.youtube-nocookie.com"].includes(urlObject.host)) return false; 
+    if(!["www.youtube.com","www.youtube-nocookie.com"].includes(url.host)) return false; 
```

Comment:
Why did you change it from the two variables of different types having different names?

Reply:
"this was done to allow url objects to be parsed to it directly but never got used"

Reply:
Changed